management-api-server/doc/openapi.json

```diff
@@ -7,12 +7,16 @@
       "url" : "http://www.apache.org/licenses/LICENSE-2.0.html"
```

Comment:
Another good question. It's a little of both.

First, the [plugin](https://github.com/k8ssandra/management-api-for-apache-cassandra/blob/master/management-api-server/pom.xml#L291-L312) is configured to run at compile time during the resolve phase so that all of the _server_ module code is compiled before trying to make sense of the annotations. The annotations on each of the methods in the server maven sub-module (the one that defines all the endpoints) drives the bulk of the doc generation. However, the jaxrs Application setup in [ManagementApplication](https://github.com/k8ssandra/management-api-for-apache-cassandra/blob/master/management-api-server/src/main/java/com/datastax/mgmtapi/ManagementApplication.java#L28-L76) doesn't quite setup everything needed for the doc generation to happen via annotations alone. So, the [plugin](https://github.com/k8ssandra/management-api-for-apache-cassandra/blob/master/management-api-server/pom.xml#L300) is also configured to merge the auto-generated doc with a static file that provides the rest of the bits [here](https://github.com/k8ssandra/management-api-for-apache-cassandra/blob/master/management-api-server/src/main/resources/openapi-configuration.json).

Ideally we would have the doc completely auto-generated, but that is a little out of scope here. I did manage to remove the configure endpoint from the static file, so there's even less that is statically managed.

Reply:
actually, this whole doc change I think needs to be reverted and moved to the other PR. Some of the changes in here reflect work done in the other PR.....

Reply:
Cool, so basically we have a bit of dodginess around the configuration endpoint and we are working around that by defining it manually, then everything else gets merged in? I can live with that for now :)

Reply:
> Cool, so basically we have a bit of dodginess around the configuration endpoint and we are working around that by defining it manually, then everything else gets merged in? I can live with that for now :)

I guess I should have read this first :smile: If this is what you would prefer, then the whole discussion above about Content-Type separation can go away for now, and I can revert that bit.

Reply:
Let's do a quick catch up tomorrow if possible just to close this out. If you have a solution that you're happier with then I think keep it, but I just want to make sure I understand the implications of both solutions.

Reply:
From discussions with @emerkle826 we are going to defer re-working this for later. For now we will continue to manually generate the OpenAPI stuff associated with these endpoints rather than combining them into one endpoint.